collect/src/main/kotlin/com/certora/collect/TreapListNode.kt

```diff
@@ -430,5 +452,11 @@ internal class TreapListNode<E> private constructor(
             // Build the whole list
             return buildLowerPri(Int.MAX_VALUE, elems.next(), Random.Default.nextInt()).node
         }
+
+        internal tailrec fun <E> TreapListNode<E>?.isApproximatelySmallerThanLog2(sizeLog2: Int): Boolean = when {
+            this == null -> true
+            sizeLog2 <= 0 -> false
```

Comment:
when could this ever be *less* than zero? Seems like we should throw on this behavior?

Reply:
Good point!